### PR TITLE
Rename iPhoneOS to iOS.

### PR DESCRIPTION
**Pull request by bruvzg**
Part of https://github.com/godotengine/godot/pull/63225

### CODE REVIEW

File: SConstruct [hunk 1]
```diff
@@ -122,8 +122,8 @@ if env['version'] == '3.x':
         if env['arch'] != 'armv7':
             env.Prepend(CXXFLAGS=['-fomit-frame-pointer'])
 elif env['version'] == '4.0':
-    env.Prepend(CFLAGS=['-std=gnu11'])
-    env.Prepend(CXXFLAGS=['-DVULKAN_ENABLED', '-std=gnu++17'])
+    env.Prepend(CFLAGS=['-std=gnu11', '-DVERSION_MAJOR=4'])
```
[the remark this answers is not in the document]
Comment (bruvzg) [reply SConstruct]: Indeed, it was missing `#include "core/version.h"` in a few places, removed it from the script.

File: README.md [hunk 2]
```diff
@@ -37,7 +37,7 @@ If you choose this option, you can skip next the step which generates Godot head
 - To generate Godot headers, you need to run the compilation command inside the `godot` submodule directory:
 
 ```bash
-scons platform=iphone target=debug
+scons platform=<iphone|ios> target=debug
```
Comment (akien-mga) [inline README.md:40]: I'd just use `ios` only.
Not all users are familiar with this kind of formatting to show possible arguments and might just copy paste as is and get an error.

Comment (akien-mga) [reply README.md:40]: Ah I guess it's different for `3.x` and 4.0.
Maybe:
```suggestion
# For Godot 3.x:
scons platform=iphone target=debug
# For Godot 4.x:
scons platform=ios target=debug
```

**Review by akien-mga — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (bruvzg) [reply README.md]: Updated:
Screenshot 2022-07-21 at 11 34 28

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**